src/pollers/mod.rs

```diff
@@ -60,21 +64,84 @@ pub struct ServerGatewayOptions {
 
     /// Timeout for long polls (polling of task queues)
     pub long_poll_timeout: Duration,
+
+    /// If specified, use TLS as configured by the [TlsCfg] struct
+    pub tls_cfg: Option<TlsConfig>,
+}
+
+/// Configuration options for TLS
+#[derive(Clone, Debug)]
+pub struct TlsConfig {
+    /// A location to a file representing the root CA certificate used by the server. If this is
+    /// set, core will attempt to use TLS when connecting to the server. If `client_tls_config` is
+    /// also set, then core will attempt to use mTLS.
+    pub server_root_ca_cert: PathBuf,
+    /// The domain/dns name that the server's certificate uses
+    pub domain: String,
```

Comment:
AFAIU, you're not supposed to set it a custom domain in production, the domain should be derived from the server address.
In nodejs this is optional: https://gist.github.com/bergundy/32f181f056760928cf934d2d4f695e33#file-client-js-L14

https://grpc.github.io/grpc/core/group__grpc__arg__keys.html#ga218bf55b665134a11baf07ada5980825